### PR TITLE
Use new Jira users/search endpoint where possible

### PR DESCRIPTION
**Pull request by eliwind**

### CODE REVIEW

File: jf_agent/jira_download.py [hunk 1]
```diff
@@ -273,30 +273,29 @@ def download_worklogs(jira_connection, issue_ids):
...
-        # a wildcard
-        users = _search_users(
-            jira_connection, gdpr_active, query='%', start_at=start_at, include_inactive=True
-        ) or _search_users(
-            jira_connection, gdpr_active, query='@', start_at=start_at, include_inactive=True
-        )
+        users = None
+
+        # different jira versions / different times, the way to list all users has changed. Try a few.
+        for q in [None, '', '%', '@']:
```
Comment (glenn-barnett) [inline jf_agent/jira_download.py]: should this `for` loop be outside the `while True`?  so it paginates through `q=None`, then through `q=''`, then `q='%'` etc?

File: jf_agent/jira_download.py [hunk 2]
```diff
@@ -273,30 +273,29 @@ def download_worklogs(jira_connection, issue_ids):
...
+def _jira_user_key(user_dict):
+    return user_dict.get('key', user_dict['accountId'])
 
 
 def _search_all_users(jira_connection, gdpr_active):
     jira_users = {}
     start_at = 0
+
+    # iterate through pages of results
     while True:
```
Comment (jnicoleoliveira) [inline jf_agent/jira_download.py]: Do you need this while true?

**Comment by jnicoleoliveira** [comment]
I approved before noticing what @glenn-barnett pointed out. I think that while True should be removed.

File: jf_agent/jira_download.py (hunk 1 above)
Comment (eliwind) [reply jf_agent/jira_download.py]: I thought the code was simpler this way, but you're right it's a little wasteful to be trying all the query strings on each page iteration.

**Pull request merged**